internal/registry.go

```diff
@@ -15,130 +10,32 @@ type internalSender interface {
 	SendDeltaCounter(name string, value float64, source string, tags map[string]string) error
 }
 
-// metric registry for internal metrics
-type MetricRegistry struct {
-	source       string
-	prefix       string
-	tags         map[string]string
-	reportTicker *time.Ticker
-	sender       internalSender
-	done         chan struct{}
-
-	mtx     sync.Mutex
-	metrics map[string]interface{}
-}
-
-type RegistryOption func(*MetricRegistry)
-
-func SetSource(source string) RegistryOption {
-	return func(registry *MetricRegistry) {
-		registry.source = source
-	}
-}
-
-func SetInterval(interval time.Duration) RegistryOption {
-	return func(registry *MetricRegistry) {
-		registry.reportTicker = time.NewTicker(interval)
-	}
-}
-
-func SetTags(tags map[string]string) RegistryOption {
-	return func(registry *MetricRegistry) {
-		registry.tags = tags
-	}
-}
-
-func SetTag(key, value string) RegistryOption {
-	return func(registry *MetricRegistry) {
-		if registry.tags == nil {
-			registry.tags = make(map[string]string)
-		}
-		registry.tags[key] = value
-	}
-}
-
-func SetPrefix(prefix string) RegistryOption {
-	return func(registry *MetricRegistry) {
-		registry.prefix = prefix
-	}
-}
-
-func NewMetricRegistry(sender internalSender, setters ...RegistryOption) *MetricRegistry {
-	registry := &MetricRegistry{
-		sender:       sender,
-		metrics:      make(map[string]interface{}),
-		reportTicker: time.NewTicker(time.Second * 60),
-		done:         make(chan struct{}),
-	}
-	for _, setter := range setters {
-		setter(registry)
-	}
-	return registry
-}
-
-func (registry *MetricRegistry) NewCounter(name string) *MetricCounter {
-	return registry.getOrAdd(name, &MetricCounter{}).(*MetricCounter)
-}
-
-func (registry *MetricRegistry) NewDeltaCounter(name string) *DeltaCounter {
-	return registry.getOrAdd(name, &DeltaCounter{MetricCounter{}}).(*DeltaCounter)
-}
-
-func (registry *MetricRegistry) NewGauge(name string, f func() int64) *FunctionalGauge {
-	return registry.getOrAdd(name, &FunctionalGauge{value: f}).(*FunctionalGauge)
+type Incrementer interface {
+	Inc()
 }
 
-func (registry *MetricRegistry) NewGaugeFloat64(name string, f func() float64) *FunctionalGaugeFloat64 {
-	return registry.getOrAdd(name, &FunctionalGaugeFloat64{value: f}).(*FunctionalGaugeFloat64)
-}
```

Comment:
The code below replaces the `MetricRegistry` struct with an interface instead. This is what allows the `configuration` struct field named `registry` to return either (1) a `RealMetricRegistry` type or a (2) `NoOpRegistry` type.

Reply:
I'd like to rename this to `InternalMetricRegistry` or `InternalSDKMetricRegistry` and also rename the structs that implement it so that they align with that.

Reply:
I chose `InternalSDKMetricRegistry` since I think the word "SDK" may help with understanding the significance of these metrics.